iterative/gcp/provider.go

```diff
@@ -324,7 +324,27 @@ func getProjectService() (string, *gcp_compute.Service, error) {
 	}
 
 	if credentials.ProjectID == "" {
-		return "", nil, errors.New("Couldn't extract the project identifier from the given credentials!")
+		// 	Coerce Credentials to handle GCP OIDC auth
+		//	Common ProjectID ENVs:
+		//		https://github.com/google-github-actions/auth/blob/b05f71482f54380997bcc43a29ef5007de7789b1/src/main.ts#L187-L191
+		//		https://github.com/hashicorp/terraform-provider-google/blob/d6734812e2c6a679334dcb46932f4b92729fa98c/google/provider.go#L64-L73
+		coercedProjectID := utils.MultiEnvLoadFirst([]string{
+			"CLOUDSDK_CORE_PROJECT",
+			"CLOUDSDK_PROJECT",
+			"GCLOUD_PROJECT",
+			"GCP_PROJECT",
+			"GOOGLE_CLOUD_PROJECT",
+			"GOOGLE_PROJECT",
+		})
+		if coercedProjectID == "" {
+			// last effort to load
+			fromCredentialsID, err := utils.GCPCoerceOIDCCredentials(credentials.JSON)
```

Comment:
It's already too long, but should it be a bit more explicit?
```suggestion
			fromCredentialsID, err := utils.getGCPProjectIDfromOIDCCredentials(credentials.JSON)
```

Reply:
I like explicit, would need that Golang "start func with a capitol letter for 'public'"

Reply:
Oops! Edited, I forgot it was being used from another package. 🙈 Should this function live perhaps under `terative/gcp/provider.go` instead?